src/js/export.js

```diff
@@ -0,0 +1,212 @@
+(function (){
+  "use strict";
+
+  window.export = {
+    toFile:exportCollectionToFile
+  }
+
+  var modal;
+  var cID;
+  var cName;
+  var filename;
+  var CSVDelimiter;
+  var leafDelimiter;
+
+  var delimiters = [
+      {value: ',' , display: 'Comma (,)'},
+      {value: ';' , display: 'Semi-Colon (;)'},
+      {value: ':' , display: 'Colon (:)'},
+      {value: '|' , display: 'Pipe (|)'},
+      {value: '^' , display: 'Caret (^)'},
+      {value: '~' , display: 'Tilde (~)'},
+      {value: '\t' , display: 'Tab'},
+      {value: ' ' , display: 'Space'}
+  ];
+
+  function exportCollectionToFile(collID, colName){
+    cID = collID;
+    cName = colName;
+    createExportModal();
+
+    // Comma should probably not be the default delimiter
+    // since a lot of the entries contain commas.
+    document.getElementById("selectDelimiterCSV").selectedIndex = 1;
+    document.getElementById("selectDelimiterLeaf").selectedIndex = 3;
+
+    document.getElementById("exportCloseBtn").addEventListener('click', (evt) => {
+      destroy(modal);
+    }, false);
+
+    document.getElementById("exportCancelBtn").addEventListener('click', (evt) => {
+      destroy(modal);
+    }, false);
+
+    document.getElementById("exportBtn").addEventListener('click', (evt) => {
+      clearComplaintsExport();
+      filename = document.getElementById("exportFilename").value;
+      var filenameValid = isFilenameValid();
```

Comment:
Since `areDelimitersValid()` read the input values, you might as well move `filename = ..` into `isFilenameValid()`